src/network/serverpackethandler.cpp

```diff
@@ -993,17 +993,16 @@ void Server::handleCommand_Interact(NetworkPacket* pkt)
 		[5] u32 length of the next item (plen)
 		[9] serialized PointedThing
 		[9 + plen] player position information
-		actions:
-		0: start digging (from undersurface) or use
-		1: stop digging (all parameters ignored)
-		2: digging completed
-		3: place block or item (to abovesurface)
-		4: use item
-		5: rightclick air ("activate")
 	*/
-	u8 action;
+
+	InteractAction action;
+	{
+		u8 temp;
+		*pkt >> temp;
+		action = static_cast<InteractAction>(temp);
```

Comment:
this is pretty ugly but I don't have a better solution.

Reply:
I couldn't think of anything better as well. Overloading `>>` for `InteractAction` is overkill, as the enum is passed to the operator only once in the whole code-base.

Also, `action = (u8)temp;` is more concise, but it's better to use `static_cast` instead of plain C-style casts wherever possible (I think).

Reply:
Wouldn't `*pkt >> (u8)action;` work?

Reply:
It doesn't. Remember, we aren't assigning `action` to `pkt` (i.e. `pkt = (u8)action`), so casting won't work. :)

Here's the error message, btw:
```
cannot bind non-const lvalue reference of type ‘irr::u8& {aka unsigned char&}’ to an rvalue of type ‘irr::u8 {aka unsigned char}’
  *pkt >> (u8)action;
          ^~~~~~~~~~
```

Reply:
Sorry, of course `*pkt >> (u8)action;` doesn't work, we have to cast to a reference: `*pkt >> (u8 &)action;` does work (I've tested).

Reply:
Oh neat. I'll try it out.

Reply:
It works!